### PR TITLE
cleanup: Change header sizes and update unneded H2 titles

### PR DESCRIPTION
**Pull request by dandhlee**
CC @jennyrosewood

This PR addresses the following:
- stripped `Example` and various other h4/h5 headers of the header and simply change them to \<p> elements
- Eliminated some h2 headers with little to no context, moved some header sizes up/down to better organize the right-nav as well

Updates internally filed issue

### CODE REVIEW

File: third_party/docfx/templates/devsite/token.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
   "overrides": "Overrides",
   "implements": "Implements",
   "remarks": "Remarks",
-  "examples": "Examples",
+  "example": "Example",
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/token.json:34]: Some of them really do have multiple examples. So, we'd need to do some sort of detection of if there is only one element (e.g. check if a second element exists).

The other headers (remarks, overrides, implements, parameters, etc.) are all plural, even if the section only has one element.

File: third_party/docfx/templates/devsite/token.json [hunk 2]
```diff
@@ -53,4 +53,4 @@
   "tip": "<h5>Tip</h5>",
   "important": "<h5>Important</h5>",
   "caution": "<h5>Caution</h5>"
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/token.json:55]: Do we also need to update these? I'm not sure where they're used.

**Review by tbpg — COMMENTED** [review]
Thank you for doing this!

File: third_party/docfx/templates/devsite/token.json (hunk 1 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/token.json:34]: Seeing that this can be done for all of the other headers, I think we should do this for all other headers as well.

For now I'll move forward and try to work this bit out. Hopefully it's not too much JS trickery :S

Comment (tbpg) [reply third_party/docfx/templates/devsite/token.json:34]: Perhaps we should update the header content in a separate PR from the header levels?

Comment (dandhlee) [reply third_party/docfx/templates/devsite/token.json:34]: Sounds good. I'll leave this out for now for this PR (but almost got it working!)

File: third_party/docfx/templates/devsite/token.json (hunk 2 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/token.json]: done!

**Comment by dandhlee** [comment]
Haven't completely gotten rid of all >h3 headers, but will be addressed in another PR along with arranging the different levels for accessibility as outlined in #159. Multiple headers to be issues in another PR as well. PTAL again!

File: third_party/docfx/templates/devsite/token.json (hunk 1 above)
Comment (tbpg) [reply third_party/docfx/templates/devsite/token.json:34]: Looks like this change is still included.

File: third_party/docfx/templates/devsite/partials/class.header.tmpl.partial [hunk 3]
```diff
@@ -120,7 +120,7 @@
 <div class="markdown level0 remarks">{{{remarks}}}</div>
 {{/remarks}}
 {{#example.0}}
-<h5 id="{{id}}_examples"><strong>{{__global.examples}}</strong></h5>
+<strong>{{__global.example}}</strong>
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/class.header.tmpl.partial:123]: s/example/examples/

File: third_party/docfx/templates/devsite/partials/class.tmpl.partial [hunk 4]
```diff
@@ -2,7 +2,6 @@
 
 {{>partials/class.header}}
 {{#children}}
-<h2 id="{{id}}">{{>partials/classSubtitle}}</h2>
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/class.tmpl.partial:5]: Not following this change -- why is this deleted?

(Not saying it's wrong to delete necessarily, but I don't understand it yet.)

File: third_party/docfx/templates/devsite/partials/namespace.tmpl.partial [hunk 5]
```diff
@@ -5,9 +5,8 @@
 <div class="markdown level0 conceptual">{{{conceptual}}}</div>
 <div class="markdown level0 remarks">{{{remarks}}}</div>
 {{#children}}
-  <h2 id="{{id}}">{{>partials/namespaceSubtitle}}</h2>
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/namespace.tmpl.partial:8]: This heading seems relevant, too?

File: third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial [hunk 6]
```diff
@@ -41,7 +41,7 @@
 <div class="markdown level0 remarks">{{{remarks}}}</div>
 {{/remarks}}
 {{#example.0}}
-<h5 id="{{id}}_examples"><strong>{{__global.examples}}</strong></h5>
+<strong>{{__global.example}}</strong>
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial:44]: s/example/examples/

File: third_party/docfx/templates/devsite/partials/uref/class.tmpl.partial [hunk 7]
```diff
@@ -2,7 +2,6 @@
 
 {{>partials/uref/class.header}}
 {{#children}}
-<h3 id="{{id}}">{{>partials/classSubtitle}}</h3>
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/uref/class.tmpl.partial:5]: Same question.

File: third_party/docfx/templates/devsite/partials/uref/namespace.examples.tmpl.partial [hunk 8]
```diff
@@ -1,10 +1,10 @@
 {{!Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.}}
 {{#example.0}}
-<h5 id="{{id}}_examples">{{__global.examples}}</h5>
+<strong>{{__global.example}}</strong>
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/uref/namespace.examples.tmpl.partial:3]: s/example/examples/

File: third_party/docfx/templates/devsite/partials/uref/namespace.tmpl.partial [hunk 9]
```diff
@@ -14,10 +14,9 @@
 {{/package.0}}
 <div class="markdown level0 remarks">{{{remarks}}}</div>
 {{#children}}
-  <h2 id="{{id}}">{{>partials/namespaceSubtitle}}</h2>
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/partials/uref/namespace.tmpl.partial:17]: Same question.

**Review by tbpg — COMMENTED** [review]
Looks like some headers were removed, but I don't quite understand why. Commented inline.

File: third_party/docfx/templates/devsite/partials/class.tmpl.partial (hunk 4 above)
Comment (tbpg) [reply third_party/docfx/templates/devsite/partials/class.tmpl.partial:5]: More thoughts: this avoids having two headers with no content in between. However, these headings provide relevant semantic meaning for the page: functions, methods, constants, variables, types, and so on. It seems confusing if we drop those headers and have the underlying elements all at the same level.

To make it explicit, this `Constants` header is being deleted, which makes the page more difficult to understand: https://github.com/googleapis/doc-templates/pull/160/files#diff-3f1b5a3a83b83d7e838c9c0763f5caa5eb1cb7fe0bc6bb624c004ae2a3ea4e5dL251

cc @jennyrosewood, please take a look.

File: third_party/docfx/templates/devsite/partials/class.header.tmpl.partial (hunk 3 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/class.header.tmpl.partial]: forgot to leave it out for this PR, done.

File: third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial (hunk 6 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/uref/class.header.tmpl.partial]: done.

File: third_party/docfx/templates/devsite/partials/uref/namespace.examples.tmpl.partial (hunk 8 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/uref/namespace.examples.tmpl.partial]: done.

File: third_party/docfx/templates/devsite/partials/class.tmpl.partial (hunk 4 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/class.tmpl.partial:5]: The headers were removed as suggested, and I can see this going either way:

Removing the headers does seem to make it quite hard to differentiate between sections. However it does help include some of the sub-headers (for example, for Go) to show up in the Table of Contents as we bumped some of them from h4 to h3. Removing the h2 headers required all the `partials/[namespace|class]Subtitle` lines to be removed.

Keeping the headers would help differentiate the sections. However that would cause some sub-headers to no longer appear in the Table of Contents as they would be pushed to h4.

Comment (tbpg) [reply third_party/docfx/templates/devsite/partials/class.tmpl.partial:5]: I would prefer to have the differentiated sections over having the sub-headers show up in the TOC. The separation is key to understanding the content on the page.

I would be OK with getting rid of the `Types` header and bumping the elements within it up one level. We'd still keep the `Variables` and `Constants` and other headers, though. That way, you would see a type and its methods/functions/etc. in the TOC.

Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/class.tmpl.partial:5]: Sounds good. Moving forward with just Types gone.

File: third_party/docfx/templates/devsite/partials/uref/namespace.tmpl.partial (hunk 9 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/uref/namespace.tmpl.partial:17]: reverted, and took out Types

File: third_party/docfx/templates/devsite/partials/uref/class.tmpl.partial (hunk 7 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/uref/class.tmpl.partial:5]: reverted as `h2`

File: third_party/docfx/templates/devsite/partials/namespace.tmpl.partial (hunk 5 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/partials/namespace.tmpl.partial]: reverted.

**Comment by dandhlee** [comment]
Reverted the group headers except Types, made them as `h2` if necessary. PTAL!

**Review by tbpg — APPROVED** [review]
Go changes look good. Might want to revert the `h4` -> `strong` Parameters, etc. changes.

(Would they need to be in a `<p>`?)

**Comment by dandhlee** [comment]
Will open a separate PR to address those that have turned into `<p><strong>` elements.
